cfme/common/provider.py

```diff
@@ -1205,6 +1208,28 @@ def get_template_guids(self, template_dict):
                         result_list.append(inner_tuple)
         return result_list
 
+    def run_smartstate_analysis_from_provider(self, datastores, wait_for_task_result=False):
+        """ Runs smartstate analysis on this host
+
+        Note:
+            The host must have valid credentials already set up for this to work.
+        """
+        view = navigate_to(self.provider, 'DatastoresOfProvider')
```

Comment:
Why is this referencing `self.provider` when its a `BaseProvider` method?

To me this should be a DatastoresCollection method, since it operates on a list of Datastore entities.

It won't handle being called with datastores that have different providers either, resulting in a ValueError.  That limitation should at least be documented, if not explicitly checked and handled, grouping datastores by their provider in order to run the core of  the function.

Reply:
I am not sure why did I put it in provider. I moved this to `datastores.py`. Seem like there is no problem here.